lua/packer/luarocks.lua

```diff
@@ -0,0 +1,305 @@
+-- Add support for installing and cleaning Luarocks dependencies
+-- Based off of plenary/neorocks/init.lua in https://github.com/nvim-lua/plenary.nvim
+local a = require('packer.async')
+local jobs = require('packer.jobs')
+local log = require('packer.log')
+local result = require('packer.result')
+local util = require('packer.util')
+
+local fmt = string.format
+local async = a.sync
+local await = a.wait
+
+local lua_version = nil
+if jit then
+  local jit_version = string.gsub(jit.version, 'LuaJIT ', '')
+  lua_version = {lua = string.gsub(_VERSION, 'Lua ', ''), jit = jit_version, dir = jit_version}
+end
+
+local cache_path = vim.fn.stdpath('cache')
+local rocks_path = util.join_paths(cache_path, 'packer_hererocks')
+local hererocks_file = util.join_paths(rocks_path, 'hererocks.py')
+local hererocks_install_dir = util.join_paths(rocks_path, lua_version.dir)
+local _hererocks_setup_done = false
+local function hererocks_is_setup()
+  if _hererocks_setup_done then return true end
+  await(a.main)
+  _hererocks_setup_done = vim.fn.isdirectory(util.join_paths(hererocks_install_dir, 'lib')) > 0
+  return _hererocks_setup_done
+end
+
+local function hererocks_installer(disp)
+  return async(function()
+    local hererocks_url = 'https://raw.githubusercontent.com/luarocks/hererocks/latest/hererocks.py'
+    local hererocks_cmd
+    await(a.main)
+    vim.fn.mkdir(rocks_path, 'p')
+    if vim.fn.executable('curl') > 0 then
+      hererocks_cmd = 'curl ' .. hererocks_url .. ' -o ' .. hererocks_file
+    elseif vim.fn.executable('wget') > 0 then
+      hererocks_cmd = 'wget ' .. hererocks_url .. ' -O ' .. hererocks_file .. ' --verbose'
+    else
+      return result.err('"curl" or "wget" is required to install hererocks')
+    end
+
+    if disp ~= nil then disp:task_start('luarocks', 'installing hererocks...') end
+
+    local output = jobs.output_table()
+    local callbacks = {
+      stdout = jobs.logging_callback(output.err.stdout, output.data.stdout, nil, disp, 'luarocks'),
+      stderr = jobs.logging_callback(output.err.stderr, output.data.stderr)
+    }
+
+    local opts = {capture_output = callbacks}
+    local r = await(jobs.run(hererocks_cmd, opts)):map_err(
+                function(err)
+        return {msg = 'Error installing hererocks', data = err, output = output}
+      end)
+
+    local luarocks_cmd = 'python ' .. hererocks_file .. ' --verbose -j ' .. lua_version.jit
```

Comment:
may want to be able to configure `python` command somehow, in case it's not in path or need to specify certain version, etc.